raft/log.go

```diff
@@ -627,20 +626,21 @@ func (l *Log) Leave() error {
 // Returns once the state has transitioned to the initial state passed in.
 func (l *Log) startStateLoop(closing <-chan struct{}, state State) {
 	l.wg.Add(1)
-	go l.stateLoop(closing, state)
+	stateChanged := make(chan struct{})
+	go l.stateLoop(closing, state, stateChanged)
 
 	// Wait until state change.
-	for {
-		if l.state == state {
-			break
-		}
-		runtime.Gosched()
-	}
+	<-stateChanged
 }
 
 // stateLoop runs in a separate goroutine and runs the appropriate state loop.
-func (l *Log) stateLoop(closing <-chan struct{}, state State) {
+func (l *Log) stateLoop(closing <-chan struct{}, state State, stateChanged chan struct{}) {
 	defer l.wg.Done()
+
+	l.Logger.Printf("log state change: %s => %s", l.state, state)
```

Comment:
With these two lines in place, you should remove them from the `for` loop, right?